gpu/gpudraw/drawused.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2024 Cogent Core. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package gpudraw
+
+import (
+	"image"
+	"image/draw"
+
+	"cogentcore.org/core/gpu"
+	"cogentcore.org/core/math32"
+)
+
+// UseGoImage uses the given Go image.Image as the source image
+// for the next Draw operation.
+func (dw *Drawer) UseGoImage(img image.Image) {
+	dw.Lock()
+	defer dw.Unlock()
+	dw.curImageSize = img.Bounds().Size()
+	tvv := dw.getNextTextureValue()
+	tvv.SetFromGoImage(img, 0)
+}
+
+// getNextTextureValue gets the next Texture image value, for Use
+// methods.
+func (dw *Drawer) getNextTextureValue() *gpu.Value {
+	sy := dw.Sys
+	tvr := sy.Vars.VarByName(1, "TexSampler")
+	ni := dw.curTexIdx
+	dw.curTexIdx++
+	nv := len(tvr.Values.Values)
+	if ni >= nv {
+		tvr.SetNValues(&sy.Device, nv+AllocChunk)
+	}
+	return tvr.Values.Values[ni]
+}
+
+// SetFrameTexture sets given gpu.Framebuffer image as a drawing source at index,
+// used in subsequent Draw methods.  Must have already been configured to fit!
+func (dw *Drawer) SetFrameTexture(idx int, fbi any) {
+	// todo use above api
+	// fb := fbi.(*gpu.Framebuffer)
+	// if fb == nil {
+	// 	return
+	// }
+	// dw.Lock()
+	// _, tx, _ := dw.Sys.Vars().ValueByIndexTry(0, "Tex", idx)
+	// if fb.Format.Size != tx.Texture.Format.Size {
+	// 	dw.Unlock()
+	// 	dw.ConfigTexture(idx, &fb.Format)
+	// 	dw.Lock()
+	// }
+	// cmd := dw.Sys.MemCmdStart()
+	// fb.CopyToTexture(&tx.Texture.Texture, dw.Sys.Device.Device, cmd)
+	// dw.Sys.MemCmdEndSubmitWaitFree()
+	// dw.Unlock()
+}
+
+// CopyUsed copies the current Use* texture to render target.
+// Must have called StartDraw and a Use* method first!
+//   - dp is the destination point.
+//   - src is the source image. If an image.Uniform, fast Fill is done.
+//   - sr is the source region, if zero full src is used.
+//   - op is the drawing operation: Src = copy source directly (blit),
+//     Over = alpha blend with existing.
+//   - flipY = flipY axis when drawing this image.
+func (dw *Drawer) CopyUsed(dp image.Point, sr image.Rectangle, op draw.Op, flipY bool) {
```

Comment:
These functions used to not have `Used` suffixes, right? Can you explain the reasoning for that change? If you want to make this change, you also have to update `system.Drawer` accordingly.

Reply:
see the README   also I forgot to check in draw.go